mash_client/cli.py

```diff
@@ -159,42 +160,51 @@ def delete(context, job_id):
 @click.group()
 def account():
     """
-    Handle mash account requests.
+    Submit account requests to the MASH server.
     """
 
 
-@click.command(name='delete')
+@click.command(
+    name='delete', context_settings=dict(token_normalize_func=str.lower)
+)
 @click.option(
     '--delete',
     is_flag=True,
     callback=abort_if_false,
     expose_value=False,
+    help='Force deletion without prompt.',
     prompt='Are you sure you want to delete account?'
 )
-@click.argument(
-    'account_name',
-    type=click.STRING
+@click.option(
+    '--name',
+    type=click.STRING,
+    required=True,
+    help='Name for the account to delete.'
 )
-@click.argument(
-    'provider',
-    type=click.STRING
+@click.option(
+    '--provider',
```

Comment:
Sorry we are going to have to switch this to "--framework", "--platform" or "--cloud" I know that is a big mess but "--provider" is simply not correct and we need to fix it.

The provider is AWS the framework/platform/cloud of platform is ec2 for Azure the provider is Microsoft. And what would we do when we end up with OpenStack, i.e. mash needs to operate against our own SOC install, then the provider would be SUSE.

For OpenStack the problem of course get broader as we could end up with multiple providers T-Systems in Germany operates a framework based on OpenStack and so do other providers. But the underlying implementation with the OpneStack is the same for everyone. So the way to handle it would be "--framework openstack"  and the provider, i.e. end point would be encoded in the account information for an OpenStack account